libs/traces/src/test/java/de/westnordost/osmapi/traces/GpsTracesApiTest.java

```diff
@@ -209,6 +213,37 @@ public class GpsTracesApiTest
 		trace = privilegedApi.get(id);
 		assertNull(trace);
 	}
+
+	@Test public void createFromGpxFileGetUpdateDelete() throws IOException
+	{
+		List<String> tags = new ArrayList<>();
+		tags.add("a tag, another");
+		ClassLoader classLoader = getClass().getClassLoader();
+		File gpxFile = new File(classLoader.getResource("track.gpx").getFile());
+		InputStream is = new FileInputStream(gpxFile);
+
+		long id = privilegedApi.create("gpx file test case", Visibility.PRIVATE, "test case from gpx file desc",
+				tags, is);
```

Comment:
Oh, don't forget to close the file input stream after it is read, otherwise the file will stay open.

Also in your own OSMTracker code of course.

I read up whether it is good practice to close the input stream already in our case in `TracesApi::create` but the general opinion is that an input stream should be closed at the level where it is created. In any case, it might be  a good hint to explicitly state that in the javadoc, too.

Reply:
Thanks for the reminder to close the stream. Will add a line in the javadoc comment of the method.